packages/components/src/components/editor/condition-builder/conditionBuilderContext.tsx

```diff
@@ -0,0 +1,172 @@
+import { createContext, useContext, useState, type ReactNode } from 'react';
+import { Input } from '@/components/common/input/input';
+import type { ConditionMode } from '@/components/editor/condition-builder/conditionBuilder';
+
+export interface ConditionData {
+  argument1: string;
+  operator: Operator;
+  argument2: string;
+  logicalOperator: LogicOperator;
+}
+
+export interface ConditionGroupData {
+  conditions: ConditionData[];
+  logicalOperator: LogicOperator;
+}
+
+export type Operators = {
+  'equal to': string;
+  'not equal to': string;
+  'is true': string;
+  'is false': string;
+  'is empty': string;
+  'is not empty': string;
+  'less than': string;
+  'greater than': string;
+  'less or equal to': string;
+  'greater or equal to': string;
+};
+
+export type Operator = keyof Operators;
+
+export type LogicOperators = {
+  and: string;
+  or: string;
+};
+
+export type LogicOperator = keyof LogicOperators;
+
+export interface ConditionBuilderProviderProps {
+  typeOptions: Operators;
+  logicalOperatorOptions: LogicOperators;
+  generateConditionString: (conditionMode: ConditionMode, conditionGroups: ConditionGroupData[]) => string;
```

Comment:
I prefer the Array<ConditionGroupData> syntax over the ConditionGroupData[] syntax...

Reply:
So should I change all the type definitions where i put bla[] instead of Array<bla>?

Reply:
If you want. I just wanted to mention it. 
https://www.totaltypescript.com/array-types-in-typescript